lib/src/Base/Geom/LevelSetMesher.cxx

```diff
@@ -177,17 +181,41 @@ Mesh LevelSetMesher::build(const LevelSet & levelSet,
     // If enough vertices, keep the simplex and flag all the vertices
     if (numGood > 0)
     {
-      goodSimplices.add(currentSimplex);
+      // If we draw the internal simplices we add the current simplex
+      // as soon as some of its vertices are inside of the level set
+      if (!surface) goodSimplices.add(currentSimplex);
+      // If we draw only the surface simplices we add the current simplex
+      // either if only some of its vertices are inside of the level set
+      // or on the bounding interval
+      if (surface)
+	{
+	  if (numGood <= dimension) goodSimplices.add(currentSimplex);
+	  else
+	    {
+	      Bool addSimplex = false;
+	      Point localVertex(dimension);
+	      // For each vertex of the current simplex
+	      for (UnsignedInteger j = 0; j <= dimension && !addSimplex; ++j)
+		{
+		  const UnsignedInteger index = currentSimplex[j];
+		  localVertex = boundingVertices[index];
```

Comment:
There is still a Point copy here, so your code is equivalent to using
```
  const Point localVertex(boundingVertices[currentSimplex[j]]);
```
which is more readable IMHO.
But it would be better to not use localVertex at all, and instead check `boundingVertices(index, k)`.

Reply:
ok I will do that as well as all the other suggestions after the merge of PR#654.